packages/trackerless-network/src/logic/StreamrNode.ts

```diff
@@ -0,0 +1,416 @@
+    acceptProxyConnections?: boolean
+}
+
+export class StreamrNode extends EventEmitter<Events> {
+    private P2PTransport?: ITransport
+    private connectionLocker?: ConnectionLocker
+    private layer0?: ILayer0
+    private streamEntryPointDiscovery?: StreamEntryPointDiscovery
+    private readonly metricsContext: MetricsContext
+    private readonly metrics: Metrics
+    public config: StreamrNodeConfig
+    private readonly streams: Map<string, StreamObject>
+    private readonly knownStreamEntryPoints: Map<string, PeerDescriptor[]> = new Map()
+    protected extraMetadata: Record<string, unknown> = {}
+    private started = false
+    private destroyed = false
+
+    constructor(config: StreamrNodeConfig) {
+        super()
+        this.config = config
+        this.streams = new Map()
+        this.metricsContext = config.metricsContext ?? new MetricsContext()
+        this.metrics = {
+            publishMessagesPerSecond: new RateMetric(),
+            publishBytesPerSecond: new RateMetric()
+        }
+        this.metricsContext.addMetrics('node', this.metrics)
+    }
+
+    async start(startedAndJoinedLayer0: ILayer0, transport: ITransport, connectionLocker: ConnectionLocker): Promise<void> {
+        if (this.started || this.destroyed) {
+            return
+        }
+        logger.info(`Starting new StreamrNode with id ${keyFromPeerDescriptor(startedAndJoinedLayer0.getPeerDescriptor())}`)
+        this.started = true
+        this.layer0 = startedAndJoinedLayer0
+        this.P2PTransport = transport
+        this.connectionLocker = connectionLocker
+        this.streamEntryPointDiscovery = new StreamEntryPointDiscovery({
+            ownPeerDescriptor: this.getPeerDescriptor(),
+            streams: this.streams,
+            getEntryPointData: (key) => this.layer0!.getDataFromDht(key),
+            getEntryPointDataViaPeer: (peerDescriptor, key) => this.layer0!.findDataViaPeer(peerDescriptor, key),
+            storeEntryPointData: (key, data) => this.layer0!.storeDataToDht(key, data),
+            deleteEntryPointData: (key) => this.layer0!.deleteDataFromDht(key)
+        })
+        cleanUp = () => this.destroy()
+    }
+
+    async destroy(): Promise<void> {
+        if (!this.started || this.destroyed) {
+            return
+        }
+        logger.trace('Destroying StreamrNode...')
+        this.destroyed = true
+        this.streams.forEach((stream) => {
+            stream.layer2.stop()
+            stream.layer1?.stop()
+        })
+        await this.streamEntryPointDiscovery!.destroy()
+        this.streams.clear()
+        this.removeAllListeners()
+        await this.layer0!.stop()
+        await this.P2PTransport!.stop()
+        this.layer0 = undefined
+        this.P2PTransport = undefined
+        this.streamEntryPointDiscovery = undefined
+        this.connectionLocker = undefined
+    }
+
+    subscribeToStream(streamPartID: string): void {
+        if (!this.streams.has(streamPartID)) {
```

Comment:
Should we rename the methods which operate on StreamPart instead of Stream? E.g. `subscribeToStreamPart`, `publishToStreamPart`, `unsubscribeFromStreamPart`, `leaveStreamPart` and `createStreamPart`?